src/Microsoft.AspNetCore.BuildTools.ApiCheck/build/Microsoft.AspNetCore.BuildTools.ApiCheck.targets

```diff
@@ -0,0 +1,30 @@
+﻿<!--
+  WARNING: These targets are intended for building Microsoft's ASP.NET Core repos and are not intended
+  for use outside of Microsoft.
+-->
+<Project>
+  <Target Name="ApiCheck" Condition=" '$(EnableApiCheck)' == 'true' AND $(IsPackable) == 'true' ">
```

Comment:
Pack and api check are separate and shouldn't be intermingled. It's possible to produce a binary can end up in a nuget package even when IsPackable is false. EnableApiCheck should be sufficient to disable the target.

The target can use existence of baseline.json files to avoid running. 
The target can use the `IsTestProject` property to suppress the warning on test projects.